src/otx/engine/hpo/hpo_api.py

```diff
@@ -175,12 +178,7 @@ def _get_default_search_space(self) -> dict[str, Any]:
         """Set learning rate and batch size as search space."""
         search_space = {}
 
-        optimizer_conf = self._engine.model.optimizer_callable
-
-        if not callable(optimizer_conf):
-            raise TypeError(optimizer_conf)
-
-        search_space["model.optimizer_callable.keywords.lr"] = self._make_lr_search_space(optimizer_conf)
+        search_space["model.optimizer_callable.lr"] = self._make_lr_search_space(self._engine.model.optimizer_callable)
```

Comment:
```suggestion
        search_space["model.optimizer_callable.optimizer_kwargs.lr"] = self._make_lr_search_space(self._engine.model.optimizer_callable)
```
HPO will change HP based on key. So, `optimizer_kwargs` needs to be passed.

Reply:
Fixed in https://github.com/openvinotoolkit/training_extensions/pull/3258/commits/2a183509b04994e73991341610f12b32d1b0489c